### PR TITLE
feature/backend/admin_search

### PR DESCRIPTION
**Pull request by helloukey**
Added fix for:

1) User list: pagination is not proper
2) User search: pagination details are not being returned
3) Session search: search based on user ID/name, pagination
4) Session list: check pagination
5) Request message: Don’t return it

### CODE REVIEW

File: backend/controllers/adminController.js [hunk 1]
```diff
@@ -187,19 +196,24 @@ const sessionList = async (req, res) => {
 */
 const sessionSearch = async (req, res) => {
   try {
-    const { page = 1, limit = 20 } = req.query;
+    const { page = 1, limit = 20, searchString } = req.query;
 
-    const sessions = await Session.find({})
+    // find based on _id or firstName or lastName
+    const sessions = await Session.find({$or: [{"_id": searchString}, {$text: {$search: searchString}}]})
```
Comment (nikhil-g777) [inline backend/controllers/adminController.js:202]: Why did you add `{requestMessage: 0}` in some of these search filters ??

**Review by nikhil-g777 — COMMENTED** [review]
@helloukey , fixed some things related to pagination, have a look at it.

File: backend/controllers/adminController.js (hunk 1 above)
Comment (helloukey) [reply backend/controllers/adminController.js:202]: To prevent the **requestMessage** field from returning.

File: backend/controllers/adminController.js [hunk 2]
```diff
@@ -85,14 +85,19 @@ const userSearch = async (req, res) => {
...
       .skip((page - 1) * limit);
 
+    // get count of all users with the search string, this is used for pagination.
+    const count = await User.countDocuments({ searchString });
+
     // return response with posts, total pages, and current page
     return res.json({
       success: true,
       users,
+      totalPages: Math.ceil(count / limit),
```
[the remark this answers is not in the document]
Comment (helloukey) [reply backend/controllers/adminController.js:99]: Ah right!
Currently, count only accepts the userType query. I'll make some changes and push it.
